cmd/influxd/launcher/launcher.go

```diff
@@ -1036,48 +1036,13 @@ func (m *Launcher) run(ctx context.Context) (err error) {
 
 	// NATS streaming server
 	natsOpts := nats.NewDefaultServerOptions()
-
-	// Welcome to ghetto land. It doesn't seem possible to tell NATS to initialise
-	// a random port. In some integration-style tests, this launcher gets initialised
-	// multiple times, and sometimes the port from the previous instantiation is
-	// still open.
-	//
-	// This atrocity checks if the port is free, and if it's not, moves on to the
-	// next one. This best-effort approach may still fail occasionally when, for example,
-	// two tests race on isAddressPortAvailable.
-	var total int
-	for {
-		portAvailable, err := isAddressPortAvailable(natsOpts.Host, natsOpts.Port)
-		if err != nil {
-			return err
-		}
-		if portAvailable && natsOpts.Host == "" {
-			// Double-check localhost to accommodate tests
-			time.Sleep(100 * time.Millisecond)
-			portAvailable, err = isAddressPortAvailable("localhost", natsOpts.Port)
-			if err != nil {
-				return err
-			}
-		}
-		if portAvailable {
-			break
-		}
-
-		time.Sleep(100 * time.Millisecond)
-		natsOpts.Port++
-		total++
-		if total > 50 {
-			return errors.New("unable to find free port for Nats server")
-		}
-	}
+	natsOpts.Port = nats.RandomPort
```

Comment:
If I'm understanding correctly, this value will be a `const` for any given program run. Will reusing the value cause the problems described in the comment, where the port stays open between tests and causes conflicts?

Reply:
By using the random port constant, it instructs NATS to use the system to find an available port. After calling Open, `natsOpt.Port` will contain the newly assigned port, that we can use in the rest of the program.

It is utilising this feature of nats:

https://github.com/influxdata/nats-streaming-server/commit/b9bd90b67451003a25b814c54c933f1d9eab7163

which I backported to v0.11.2:

https://github.com/influxdata/nats-streaming-server/commit/c277f7560803c533510da700acca623c2fe0d6c1